### PR TITLE
Add a test helper for creating a dummy Credentials

### PR DESCRIPTION
**Pull request by ysaito1001**
## Motivation and Context
This is a refactoring PR affecting only tests, in response to https://github.com/awslabs/smithy-rs/pull/2122#discussion_r1056499862.

## Description
We have various places in testing that create dummy `Credentials` such as
- `Credentials::new("ANOTREAL", "notrealrnrELgWzOk3IfjzDKtFBhDby", Some("notarealsessiontoken".to_string()), None, "test")`
- `Credentials::new("akid", "secret", None, None, "test")`
- `Credentials::new("example", "example", None, None, "example")`

These essentially mean "it does not matter". To consolidate all of these instances, a test helper `aws_credential_types::Credentials::for_tests` has been introduced whose implementation is taken from the first bullet point above.

We intentionally did not replace certain occurrences of dummy `Credentials::new` with `Credentials::for_tests`:
- Those that appear in code examples in rustdoc (it makes more sense for the customers to see `new` rather than `for_tests`)
- Those that need to configure the expiration time

## Testing
- [x] Passed all tests in CI

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/rust-runtime/aws-credential-types/src/credentials_impl.rs [hunk 1]
```diff
@@ -140,6 +140,21 @@ impl Credentials {
...
+    #[doc(hidden)]
+    pub fn for_tests() -> Self {
+        Self::new(
+            "ANOTREAL",
+            "notrealrnrELgWzOk3IfjzDKtFBhDby",
+            Some("notarealsessiontoken".to_string()),
+            None,
+            "test",
+        )
+    }
```
Comment (rcoh) [inline aws/rust-runtime/aws-credential-types/src/credentials_impl.rs:156]: I don't think this should be doc-hidden but we should feature gate it behind `test-util`. You can see how we handle this for `aws-smithy-client` here: https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/s3/Cargo.toml#L100-L103

File: aws/rust-runtime/aws-sig-auth/src/middleware.rs [hunk 2]
```diff
@@ -270,6 +270,7 @@ mod test {
                 .apply(req.try_clone().expect("can clone"))
                 .expect_err("no cred provider"),
         );
+        // use `new` instead of `for_tests` because the presence of a session_token will render `auth_header` below `Sensitive`.
```
Comment (rcoh) [inline aws/rust-runtime/aws-sig-auth/src/middleware.rs:273]: you can use `unwrap().to_string()` to debug IIRC

**Review by rcoh — COMMENTED** [review]
Thank you so much for doing this!!

File: aws/sdk/integration-tests/s3/tests/query-strings-are-correctly-encoded.rs [hunk 3]
```diff
@@ -15,13 +15,7 @@ use std::time::{Duration, UNIX_EPOCH};
...
     let (conn, rcvr) = capture_request(None);
     let sdk_config = SdkConfig::builder()
-        .credentials_provider(SharedCredentialsProvider::new(Credentials::new(
-            "ANOTREAL",
-            "notrealrnrELgWzOk3IfjzDKtFBhDby",
-            Some("notarealsessiontoken".to_string()),
-            None,
-            "test",
-        )))
+        .credentials_provider(SharedCredentialsProvider::new(Credentials::for_tests()))
```
Comment (Velfi) [inline aws/sdk/integration-tests/s3/tests/query-strings-are-correctly-encoded.rs:18]: What do you think of adding a method to `SharedCredentialsProvider` like so:
```suggestion
        .credentials_provider(SharedCredentialsProvider::for_tests())
```

It's just about as clear, with a little less code.

**Review by Velfi — APPROVED** [review]

File: aws/sdk/integration-tests/s3/tests/query-strings-are-correctly-encoded.rs (hunk 3 above)
Comment (ysaito1001) [reply aws/sdk/integration-tests/s3/tests/query-strings-are-correctly-encoded.rs:18]: I would normally incorporate the suggestion but knowing `SharedCredentialsProvider` will cease to exist in #2122, we should probably leave it as is.

File: aws/rust-runtime/aws-credential-types/src/credentials_impl.rs (hunk 1 above)
Comment (ysaito1001) [reply aws/rust-runtime/aws-credential-types/src/credentials_impl.rs]: Considered once the introduction of a `test` feature but did not go for it. Happy to make the change.

Comment (ysaito1001) [reply aws/rust-runtime/aws-credential-types/src/credentials_impl.rs]: Addressed in dfecd0d.

File: aws/rust-runtime/aws-sig-auth/src/middleware.rs (hunk 2 above)
Comment (ysaito1001) [reply aws/rust-runtime/aws-sig-auth/src/middleware.rs]: Great, that worked. Thanks for the suggestion. Addressed in 851abbb.

**Pull request merged**